chain/chain/src/chain.rs

```diff
@@ -544,23 +546,32 @@ impl Chain {
     // ===
     //
     // Prerequisites, guaranteed by the System:
-    // 1. Genesis is always on the Canonical Chain, only one Genesis exists and its height is `genesis_height`,
-    //    and no block with height lower than `genesis_height` exists.
-    // 2. If block A is ancestor of block B, height of A is strictly less then height of B.
-    // 3. (Property 1). An oldest block where fork is happened has the least height among all blocks on the fork.
-    // 4. (Property 2). An oldest block where fork is happened is never affected by Canonical Chain Switching
+    // 1. Genesis block is available and should not be removed by GC.
+    // 2. No block in storage except Genesis has height lower or equal to `genesis_height`.
+    // 3. There is known lowest block height (Tail) came from Genesis or State Sync.
+    //    a. Tail is always on the Canonical Chain.
+    //    b. Only one Tail exists.
+    //    c. Tail's height is higher than or equal to `genesis_height`,
+    // 4. There is known highest block height (Head).
+    //    a. Head is always on the Canonical Chain.
+    // 5. All blocks in the storage have heights in range [Tail; Head].
+    //    a. All forks end up on height of Head or lower.
+    // 6. If block A is ancestor of block B, height of A is strictly less then height of B.
+    // 7. (Property 1). A block with the lowest height among all the blocks at which the fork has started,
+    //    i.e. all the blocks with the outgoing degree 2 or more,
+    //    has the least height among all blocks on the fork.
+    // 8. (Property 2). An oldest block where fork is happened is never affected by Canonical Chain Switching
```

Comment:
Can we at least fix the grammar here? "The oldest block where the fork happened..."

Reply:
@bowenwang1996 please fix my grammar as much as possible :)